activities.el

```diff
@@ -858,44 +859,44 @@ OLDEST-POSSIBLE is the oldest age in the `vc-annotate-color-map'."
     (when-let ((activity (map-elt activities-activities name)))
       (let (data)
 	(dolist (type '(last default))
-	  (let* ((state (cl-struct-slot-value 'activities-activity type activity))
-		 (time (map-elt (activities-activity-state-etc state) 'time))
-		 (window-state (activities-activity-state-window-state state))
-		 (buffers (window-state-buffers window-state))
-		 (files (activities--map-window-state-leafs
-			 window-state
-			 (lambda (leaf)
-			   (bookmark-get-filename
-			    (activities-buffer-bookmark
-			     (map-nested-elt (cdr leaf)
-			 		     '(parameters activities-buffer))))))))
-	    (setf (alist-get type data)
-		  (list (length buffers)
-			(length (delq nil files))
-			(float-time (time-since time))))))
-	(pcase-let* ((`(,last-bufs ,last-files ,last-time) (map-elt data 'last))
-		     (`(,default-bufs ,default-files ,default-time) (map-elt data 'default))
-		     (age (min last-time default-time))
-		     (ann (format "%s:%s|%s %s:%s|%s "
-				  (propertize "buffers" 'face 'bold)
-				  (propertize (format "%2d" last-bufs) 'face 'success)
-				  (propertize (format "%-2d" default-bufs) 'face 'warning)
-				  (propertize "files" 'face 'bold)
-				  (propertize (format "%2d" last-files) 'face 'success)
-				  (propertize (format "%-2d" default-files) 'face 'warning)))
+	  (when-let ((state (cl-struct-slot-value 'activities-activity type activity)))
+	    (let* ((time (map-elt (activities-activity-state-etc state) 'time))
+		   (window-state (activities-activity-state-window-state state))
+		   (buffers (window-state-buffers window-state))
+		   (files (activities--map-window-state-leafs
+			   window-state
+			   (lambda (leaf)
+			     (bookmark-get-filename
+			      (activities-buffer-bookmark
+			       (map-nested-elt (cdr leaf)
+			 		       '(parameters activities-buffer))))))))
+	      (setf (alist-get type data)
+		    (list (length (delq nil files))
```

Comment:
`window-state-leafs` actually just pushes the results of the function it was provided, which in this case involves `bookmark-get-filename`, which returns nil for non-file buffer bookmarks.   I think it's useful to preserve the nils in case you ever want to correlate the list against other calls.